src/fsharp/fsi/fsimain.fs

```diff
@@ -316,7 +314,12 @@ let evaluateSession(argv: string[]) =
 let MainMain argv = 
     ignore argv
     let argv = System.Environment.GetCommandLineArgs()
-    use e = new SaveAndRestoreConsoleEncoding()
+    use e =
+        { new IDisposable with
+            member __.Dispose() =
+                try 
+                    Console.SetOut(Console.Out)
```

Comment:
This is curious. What is this doing?

Reply:
@cartermp The code is not correct - the original definition of `SaveAndRestoreConsoleEncoding` is:

```fsharp
type SaveAndRestoreConsoleEncoding () =
    let savedOut = System.Console.Out

    interface System.IDisposable with
        member this.Dispose() = 
            try 
                System.Console.SetOut(savedOut)
            with _ -> ()
```

which saves Console.Out for later restoral.  Since you want this PR to be highly conservative you should do the same thing

I'm a little skeptical about the code in any case - we don't call `SetOut` anywhere for FSI - but I suppose the idea is that if the user code being executed calls `SetOut` then we restore the original console output stream before exiting.  I'll leave it to experts on the console to decide if that's the right thing, I guess it's not the point of the PR either way though.

Reply:
I think I'm confused - if we're calling `SetOut` with the same value (`Console.Out`), then wouldn't it make no difference what the overall outcome is?

Reply:
Oh I think I get it now, the user could have redirected `Console.Out` to something else